### PR TITLE
Add graph page

### PR DESCRIPTION
**Pull request by Sedictious**
Simply initialize an empty page when pushing the "graph" button (uses the project button as a placeholder). This is a WIP but should be enough to give the general idea of a possible implementation.

(Solved) The commit messes up the spacing of the toolbar, but I'll fix this once I find out how Qt wspacer works exactly.

Update: 54fd499 added a simple (non yet functional) GraphDesigner which is pretty much an ExperimentDesigner with only the graph functionality present. One issue that needs to be addressed is the way attributes are handled; without an experiment we must have a way of specifying the number of attributes and perhaps other properties directly from the designer.

I opted out from making the GraphDesigner a DockWindow as I think only one graph should be able to be opened at any time and unlike the Experiment Designer, having a main page isn't necessary. But of course, I'm always open to suggestions :)

### CODE REVIEW

**Comment by cardinot** [comment]
> I opted out from making the GraphDesigner a DockWindow as I think only one graph should be able to be opened at any time and unlike the Experiment Designer, having a main page isn't necessary. But of course, I'm always open to suggestions :)

I'm ok with opening one graph/grid at a time. However, do you have plans to implement the "full inspector" widget that we discussed earlier? In this case, how would we be able to dock the inspector widget in the GraphDesigner if it's not a window? 🤔

**Comment by Sedictious** [comment]
@cardinot I was planning to have a separate ``GraphDesignerWindow`` class which would be initialized after setting up all the parameters. However, if this is the case, maybe it's worth considering remaining ``GraphDesigner`` to something like ``GraphDesignerSettings``.

**Comment by cardinot** [comment]
> @cardinot I was planning to have a separate `GraphDesignerWindow` class which would be initialized after setting up all the parameters. However, if this is the case, maybe it's worth considering remaining `GraphDesigner` to something like `GraphDesignerSettings`.

so, are you planning to open the "new graph" settings in place? (instead of a popup dialog)
I'll compile and test the existing code later, but, in this case, what will happen when I close the graph? Will you close the "graph window" and send me back to the "graph settings" page?

**Comment by Sedictious** [comment]
@cardinot I was actually planning that closing the graph would return you to the graph designer main page (the one with "New Graph" and "Open Graph")

File: src/gui/graphdesignerpage.h [hunk 1]
```diff
@@ -0,0 +1,58 @@
+/**
+*  This file is part of Evoplex.
+*
+*  Evoplex is a multi-agent system for networks.
+*  Copyright (C) 2018 - Marcos Cardinot <marcos@cardinot.net>
```
Comment (cardinot) [inline src/gui/graphdesignerpage.h]: following the license template, here we write the author's name for the file; so, feel free to replace it with your name

**Pull request merged**